File: docs/components_page/components/spinner/loading.R
library(dashBootstrapComponents)

loading_spinner <- div(
  list(
    dbcButton("Load", id = "loading-button", n_clicks = 0),
    dbcSpinner(div(id = "loading-output"))
  )
)


app$callback(
  output("loading-output", "children"),
  list(input("loading-button", "n_clicks")),
  function(n) {
    if (n > 0) {
      Sys.sleep(1)
      return(sprintf("Output loaded %d times", n))
    }
    return(sprintf("Output not reloaded yet"))
  }
)

File: docs/components_page/components/spinner/loading.jl
using DashBootstrapComponents, DashHtmlComponents

loading_spinner = html_div([
    dbc_button("Load", id = "loading-button", n_clicks = 0),
    dbc_spinner(html_div(id = "loading-output")),
]);

callback!(
    app,
    Output("loading-output", "children"),
    Input("loading-button", "n_clicks"),
) do n

    if n > 0
        sleep(1)
        return "Output loaded $n times"
    else
        return "Output not reloaded yet"
    end

end;
